src/core/tests/type_prop/nms_rotated.cpp

```diff
@@ -0,0 +1,333 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/nms_rotated.hpp"
+
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/type_prop.hpp"
+#include "openvino/op/constant.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace testing;
+
+template <class TOp>
+class NMSRotatedCommonTest : public TypePropOpTest<TOp> {};
+
+TYPED_TEST_SUITE_P(NMSRotatedCommonTest);
+
+TYPED_TEST_P(NMSRotatedCommonTest, incorrect_boxes_rank) {
```

Comment:
It's not so trivial as there is a difference in expected shape of the boxes input, and "default" inputs.
So the intention of usage of TYPED_TEST was indeed to reuse the NMS tests, anyway it can be useful for further version of the NMSRotated.